File: GoSIS/models/employee.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Employee struct {
	Id           primitive.ObjectID `bson:"_id,omitempty"`
	EmployeeId   string             `json:"employeeId"`
	FirstName    string             `json:"firstName"`
	LastName     string             `json:"lastName"`
	VacationDays int64              `json:"vacationDays"`
	PaidToDate   int64              `json:"paidToDate"`
	PaidLastYear int64              `json:"paidLastYear"`
	PayRate      float64            `json:"payRate"`
	PayRateID    int64              `json:"payRateId"`
	CreatedAt    time.Time          `json:"createdAt,omitempty"`
	UpdatedAt    time.Time          `json:"updatedAt,omitempty"`
}

type EmployeeNotID struct {
	EmployeeId   string    `bson:"employeeId" form:"employeeId"`
	FirstName    string    `bson:"firstName" form:"firstName"`
	LastName     string    `bson:"lastName" form:"lastName"`
	VacationDays int64     `bson:"vacationDays" form:"vacationDays"`
	PaidToDate   int64     `bson:"paidToDate" form:"paidToDate"`
	PaidLastYear int64     `bson:"paidLastYear" form:"paidLastYear"`
	PayRate      float64   `bson:"payRate" form:"payRate"`
	PayRateID    int64     `bson:"payRateId" form:"payRateId"`
	CreatedAt    time.Time `bson:"createdAt,omitempty"`
	UpdatedAt    time.Time `bson:"updatedAt,omitempty"`
}

File: GoSIS/faker-api/employee.py
import requests
import json
from faker import Faker

fake = Faker()

headers = {'Content-Type': 'application/json'}

n = int(input("Nhập số lượng request : "))

for i in range(n):
    data = {
        "employeeId": ''.join(fake.random_letters(length=5)),
        "firstName": fake.first_name(),
        "lastName": fake.last_name(),
        "vacationDays": fake.random_int(min=1, max=30),
        "paidToDate": fake.random_int(min=100, max=1000),
        "paidLastYear": fake.random_int(min=100, max=1000),
        "payRate": fake.random_int(min=10, max=100),
        "payRateId": fake.random_int(min=100, max=1000)
    }

    response = requests.post(
        'http://localhost:4000/api/employee', data=json.dumps(data), headers=headers)
    if response.status_code == 200:
        print(f'POST request {i+1} successful')
    else:
        print(f'Error on POST request {i+1}: {response.text}')
